packages/next/src/build/webpack/loaders/next-barrel-loader.ts

```diff
@@ -0,0 +1,73 @@
+import type webpack from 'webpack'
+
+const NextBarrelLoader: webpack.LoaderDefinitionFunction<{
+  names: string[]
+  wildcard: boolean
+}> = function (source) {
+  this.cacheable()
+  const { names, wildcard } = this.getOptions()
+
+  const matches = source.match(
+    /^([^]*)export const __next_private_export_map__ = '([^']+)'/
+  )
+
+  if (!matches) {
+    // This file isn't a barrel and we can't apply any optimizations. Let's re-export everything.
+    return `export * from ${JSON.stringify(this.resourcePath)}`
```

Comment:
should we just return `source` here? Will that break the relative imports in `this.resourcePath` if we return source directly?

Reply:
We can't because that will create duplicate instances of the original module.